pallets/permissions/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate frame_system;
+
+///! A crate that defines a simple permissions logic.
+///! Users of the create must implement the `Properties` trait on a
+///! type of their choosing in order to use this pallet properly.
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+use frame_support::{dispatch::DispatchResult, pallet_prelude::*};
+use frame_system::pallet_prelude::*;
+
+pub trait Permissions<AccountId> {
+	type Location;
+	type Role;
+	type Error;
+
+	fn clearance(location: Self::Location, who: AccountId, role: Self::Role) -> bool;
```

Comment:
I think the previous naming was simpler, "clearance" is less explicit about what it means.

```suggestion
	fn has_role(location: Self::Location, who: AccountId, role: Self::Role) -> bool;
```

Reply:
I think `clearance` describes it good with respect to the trait definition. But I understand your point. I don`t like `has_role`. What about `has_permission` to make it more obvious in contrast to `clearance`?

Reply:
Yeah I think `has_permission` is also good!